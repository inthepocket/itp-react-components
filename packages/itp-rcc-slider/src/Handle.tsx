import * as React from 'react';
import classnames from 'classnames';

import { GetHandleProps } from 'react-compound-slider/Handles/Handles';

export interface HandleStyles {
  Handle?: string;
  isHandleActive?: string;
}

export interface HandleProps {
  activeHandleID?: string;
  getHandleProps: GetHandleProps;
  id: string;
  key?: string;
  percent: number;
  prefixCss?: string;
  styles?: HandleStyles;
}

const Handle = ({
  activeHandleID,
  getHandleProps,
  id,
  percent,
  prefixCss,
  styles,
}: HandleProps) => {
  const containerClassName = styles ? styles.Handle : `${prefixCss}__Handle`;
  const activeClassName = styles ? styles.isHandleActive : `${prefixCss}__Handle--isActive`;
  const handleClassName = classnames({
    [containerClassName]: true,
    [activeClassName]: activeHandleID === id
  });
  const handleStyle = {
    left: `${percent}%`
  };

  return (
    <div
      className={handleClassName}
      role="slider"
      style={handleStyle}
      {...getHandleProps(id)}
    />
  );
};

export default Handle;
