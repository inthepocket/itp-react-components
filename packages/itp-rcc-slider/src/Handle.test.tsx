import * as React from 'react';
import { create } from 'react-test-renderer';
import Handle from './Handle';

test('Handle: default', () => {
  const tree = create((
    <Handle
      activeHandleID="foo:bar:active"
      getHandleProps={() => null}
      id="foo:bar"
      percent={75}
      prefixCss="Slider"
     />
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Handle: css module', () => {
  const tree = create((
    <Handle
      activeHandleID="foo:bar:active"
      getHandleProps={() => null}
      id="foo:bar"
      percent={75}
      styles={{
        Handle: 'cssModuleHandle',
      }}
     />
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});
