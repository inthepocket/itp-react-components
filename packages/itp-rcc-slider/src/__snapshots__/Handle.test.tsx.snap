// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handle: css module 1`] = `
<div
  className="cssModuleHandle"
  role="slider"
  style={
    Object {
      "left": "75%",
    }
  }
/>
`;

exports[`Handle: default 1`] = `
<div
  className="Slider__Handle"
  role="slider"
  style={
    Object {
      "left": "75%",
    }
  }
/>
`;
